nautobot/extras/models/customfields.py

```diff
@@ -724,7 +724,20 @@ def delete(self, *args, **kwargs):
 
         super().delete(*args, **kwargs)
 
-        delete_custom_field_data.delay(self.key, content_types)
+        # Circular Import
+        from nautobot.extras.signals import _get_user_if_authenticated, change_context_state
```

Comment:
As part of this changeset can we either:

- rename `_get_user_if_authenticated` to something non-private?
- roll it into `change_context.get_user()` as built-in functionality so we don't have to do this check all over the place?

Reply:
_get_user_if_authenticated needs the `instance` argument which is not available on the ChangeContext. 
Would it be acceptable to set `instance` to `None` when using it in `get_user()`?

Reply:
Nah, the instance is useful debugging info. Can we pass it as an optional parameter to `get_user()`?